ImageProcessing/README.md

```diff
@@ -0,0 +1,747 @@
+          "States": {
+            "Thumbnail": {
+              "Type": "Task",
+              "Resource": "REPLACE_WITH_ThumbnailFunctionArn",
+              "End": true
+            }
+          }
+        }
+      ],
+      "End": true
+    }
+	```
+	
+1. Find the line in the `CheckFaceDuplicate` state that marks it as the End state of the state machine
+
+	```JSON
+	     	 "End": true,
+
+	```
+	and replace it with
+	
+	```JSON
+      		"Next": "ParallelProcessing",
+
+	```
+	This tells AWS Step Functions if the  `CheckFaceDuplicate ` state runs successfully, go on to run the `ParallelProcessing ` state as the next step in the process. 
+
+1. At this point, your `rider-photo-state-machine.json` file should look like this (the AWS Lambda ARNs are examples): 
+	
+	<details>
+	<summary><strong>(expand to see)</strong></summary><p>
+
+	```JSON
+	{
+	  "Comment": "Rider photo processing workflow",
+	  "StartAt": "FaceDetection",
+	  "States": {
+	    "FaceDetection": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-FaceDetectionFunction-4AYSKX2EGPV0",
+	      "ResultPath": "$.detectedFaceDetails",
+	      "Next": "CheckFaceDuplicate",
+	      "Catch": [
+	        {
+	          "ErrorEquals": [
+	            "PhotoDoesNotMeetRequirementError"
+	          ],
+	          "ResultPath": "$.errorInfo",
+	          "Next": "PhotoDoesNotMeetRequirement"
+	        }
+	      ]
+	    },
+	    "PhotoDoesNotMeetRequirement": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-NotificationPlaceholderFunct-CDRLZC8BRFWP",
+	      "End": true
+	    },
+	    "CheckFaceDuplicate": {
+	      "Type": "Task",
+	      "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-FaceSearchFunction-1IT67V4J214DC",
+	      "ResultPath": null,
+	      "Next": "ParallelProcessing",
+	      "Catch": [
+	        {
+	          "ErrorEquals": [
+	            "FaceAlreadyExistsError"
+	          ],
+	          "ResultPath": "$.errorInfo",
+	          "Next": "PhotoDoesNotMeetRequirement"
+	        }
+	      ]
+	    },
+	    "ParallelProcessing": {
+	      "Type": "Parallel",
+	      "Branches": [
+	        {
+	          "StartAt": "AddFaceToIndex",
+	          "States": {
+	            "AddFaceToIndex": {
+	              "Type": "Task",
+	              "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-IndexFaceFunction-15658V8WUI67V",
+	              "End": true
+	            }
+	          }
+	        },
+	        {
+	          "StartAt": "Thumbnail",
+	          "States": {
+	            "Thumbnail": {
+	              "Type": "Task",
+	              "Resource": "arn:aws:lambda:us-west-2:012345678912:function:wild-ryde-step-module-ThumbnailFunction-A30TCJMIG0U8",
+	              "End": true
+	            }
+	          }
+	        }
+	      ],
+	      "ResultPath": "$.parallelResult",
```

Comment:
It's working for me after fixing the `ResultPath`. It didn't last night, but that's because I was omitting the `userId` attribute because I thought the instructions implied I didn't need it. This step requires an explict `userId` to be passed for persistence though. It might make sense to strike the bits about it being optional and autogenerated and make it required for each step.